miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditPoolController.java

```diff
@@ -319,4 +317,83 @@ protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) {
+
+    /** Given a bunch of strings, find the long substring that matches all of them that doesn't end in numbers or underscores. */
+    private String findCommonPrefix(String[] str) {
+      StringBuilder commonPrefix = new StringBuilder();
+
+      while (commonPrefix.length() < str[0].length()) {
+        char current = str[0].charAt(commonPrefix.length());
+        boolean matches = true;
+        for (int i = 1; matches && i < str.length; i++) {
+          if (str[i].charAt(commonPrefix.length()) != current) {
+            matches = false;
+          }
+        }
+        if (matches) {
+          commonPrefix.append(current);
+        } else {
+          break;
+        }
+      }
+      // Chew back any digits at the end
+      while (commonPrefix.length() > 0 && Character.isDigit(commonPrefix.charAt(commonPrefix.length() - 1))) {
+        commonPrefix.setLength(commonPrefix.length() - 1);
+      }
+      if (commonPrefix.length() > 0 && commonPrefix.charAt(commonPrefix.length() - 1) == '_') {
+        commonPrefix.setLength(commonPrefix.length() - 1);
+      }
+      return (commonPrefix.length() > 0) ? commonPrefix.toString() : null;
+
+    }
+
+    @Override
+    protected PoolDto createDtoFromParents(List<Long> parentIds) throws IOException {
+      PoolDto dto = new PoolDto();
+      List<Pool> parents = new ArrayList<>(poolService.listByIdList(parentIds));
+
+      if (parents.size() < 2) {
+        throw new IllegalStateException("Not enough pools to merge");
+      }
+
+      List<PlatformType> platformTypes = parents.stream().map(Pool::getPlatformType).distinct().collect(Collectors.toList());
+      if (platformTypes.size() > 1) {
+        throw new IllegalArgumentException("Cannot merge pools from multiple platforms: "
+            + String.join(", ", platformTypes.stream().map(Enum::name).toArray(CharSequence[]::new)));
+      }
+      dto.setPlatformType(platformTypes.get(0).name());
+
+      String parentString = parents.stream().map(Pool::getName).collect(Collectors.joining(", "));
+      dto.setDescription("Created from merging pools: " + parentString);
+
+      String commonPrefix = findCommonPrefix(parents.stream().map(Pool::getAlias).toArray(String[]::new));
+      if (commonPrefix != null) {
+        dto.setAlias(commonPrefix + "_POOL");
+      }
+
+      dto.setPooledElements(
+          parents.stream().flatMap(pool -> pool.getPoolableElementViews().stream()).map(Dtos::asDto).collect(Collectors.toSet()));
+      List<String> volumeUnits = parents.stream().map(Pool::getVolumeUnits).filter(units -> units != null).distinct()
+          .collect(Collectors.toList());
+      if (parents.stream().allMatch(pool -> (pool.getVolume() != null)) && volumeUnits.size() == 1) {
+        dto.setVolume(Double.toString(parents.stream().mapToDouble(Pool::getVolume).sum()));
+      }
+
+      return dto;
+    }
+
+    @Override
+    protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) throws IOException {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)